### PR TITLE
executor: let plan replayer record table tiflash replica

### LINKED ISSUE #37255: Plan replayer can't handle tiflash replica
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. Plan replayer dump query with tiflash engine like following:

```sh
mysql> explain select /*+ read_from_storage(tiflash[t]) */ * from t;
+-----------------------+----------+--------------+---------------+--------------------------------+
| id                    | estRows  | task         | access object | operator info                  |
+-----------------------+----------+--------------+---------------+--------------------------------+
| TableReader_5         | 10000.00 | root         |               | data:TableFullScan_4           |
| └─TableFullScan_4     | 10000.00 | cop[tiflash] | table:t       | keep order:false, stats:pseudo |
+-----------------------+----------+--------------+---------------+--------------------------------+
2 rows in set (0.00 sec)
```

Load result into another tidbcluster with tiflash

### 2. What did you expect to see? (Required)

Explain query will get tiflash result as well

### 3. What did you see instead (Required)

Explain query got tikv result

```sh
mysql> explain select /*+ read_from_storage(tiflash[t]) */ * from t;
+-----------------------+----------+-----------+---------------+--------------------------------+
| id                    | estRows  | task      | access object | operator info                  |
+-----------------------+----------+-----------+---------------+--------------------------------+
| TableReader_5         | 10000.00 | root      |               | data:TableFullScan_4           |
| └─TableFullScan_4     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo |
+-----------------------+----------+-----------+---------------+--------------------------------+
2 rows in set, 1 warning (0.01 sec)

```

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close #37255 

Problem Summary:

### What is changed and how it works?

let plan replayer record table tiflash replica

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)

Dump plan replayer results from query with tiflash from 1 cluster, and load it into another cluster. The query can be still executed by tiflash in another cluster.

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: executor/plan_replayer.go [hunk 1]
```diff
@@ -293,6 +310,40 @@ func dumpMeta(zw *zip.Writer) error {
...
+		dbName := model.NewCIStr(pair.DBName)
+		tableName := model.NewCIStr(pair.TableName)
+		t, err := is.TableByName(dbName, tableName)
+		if err != nil {
+			logutil.BgLogger().Warn("failed to find table info", zap.Error(err),
+				zap.String("dbName", dbName.L), zap.String("tableName", tableName.L))
+			continue
+		}
+		if t.Meta().TiFlashReplica != nil && t.Meta().TiFlashReplica.Count > 0 {
+			sql := fmt.Sprintf("SELECT TABLE_SCHEMA,TABLE_NAME,REPLICA_COUNT FROM INFORMATION_SCHEMA.TIFLASH_REPLICA WHERE TABLE_SCHEMA ='%s' AND TABLE_NAME='%s' AND REPLICA_COUNT > 0",
```
Comment (xuyifangreeneyes) [inline executor/plan_replayer.go:329]: Can we just get `TABLE_SCHEMA`, `TABLE_NAME`, `REPLICA_COUNT` from `t.Meta()`?

File: executor/plan_replayer.go [hunk 2]
```diff
@@ -348,94 +399,74 @@ func dumpVariables(ctx sessionctx.Context, zw *zip.Writer) error {
 	if err := toml.NewEncoder(vf).Encode(varMap); err != nil {
 		return errors.AddStack(err)
 	}
-	if len(recordSets) > 0 {
-		if err := recordSets[0].Close(); err != nil {
-			return err
-		}
-	}
```
Comment (xuyifangreeneyes) [inline executor/plan_replayer.go:355]: We still need to close recordSet?

File: executor/plan_replayer.go [hunk 3]
```diff
@@ -604,9 +630,49 @@ func (e *PlanReplayerLoadExec) Next(ctx context.Context, req *chunk.Chunk) error
 	return nil
 }
 
+func loadSetTiFlashReplica(ctx sessionctx.Context, z *zip.Reader) error {
+	for _, zipFile := range z.File {
+		if strings.Compare(zipFile.Name, tiFlashReplicasFile) == 0 {
+			//varMap := make(map[string]string)
```
Comment (xuyifangreeneyes) [inline executor/plan_replayer.go:636]: Remove unrelated comment.

**Review by xuyifangreeneyes — COMMENTED** [review]
rest LGTM

**Review by xuyifangreeneyes — APPROVED** [review]

**Pull request merged**